File: NETMouse - .NET release/Examples/PascalABC.NET/Matrices_2.pas
{$reference ABCNET.dll}
uses ABCNET.Utils;
uses ABCNET.Extensions;
uses System;

function Sum(self: array [,] of integer): array of integer; extensionmethod;
begin
  if self = nil then
    raise new ArgumentNullException('self');

  result := new integer[self.GetLength(0)];
  var rowsCount := self.GetLength(0);
  var colsCount := self.GetLength(1);
  for var i := 0 to rowsCount - 1 do
    for var j := 0 to colsCount - 1 do
      result[i] += self[i, j];
end;

begin
  Matr.ReadInt32(Base.ReadInt32('N:'), Base.ReadInt32('M:'), 'Элемент ({0}, {1})-ый:').Sum().Println();
  Base.ReadChar();
end.

File: NETMouse - .NET release/Examples/C#/Matrices_2.cs
using ABCNET.Extensions;
using ABCNET.Utils;
using System;

namespace TestProgram
{
    internal static class Program
    {
        public static int[] Sum(this int[,] matrix)
        {
            if (matrix == null)
                throw new ArgumentNullException(nameof(matrix));

            int[] result = new int[matrix.GetLength(0)];
            int rowsCount = matrix.GetLength(0);
            int colsCount = matrix.GetLength(1);
            for (int i = 0; i < rowsCount; i++)
                for (int j = 0; j < colsCount; j++)
                    result[i] += matrix[i, j];

            return result;
        }

        private static void Main(string[] args)
        {
            Matr.ReadInteger(Base.ReadInteger("N:"), Base.ReadInteger("M:"), "Элемент ({0}, {1})-ый:").Sum().Println();
        }
    }
}